### ISSUE TITLE
Add sigmoid_kernel Function

### ISSUE THREAD

**Issue by icfaust** [uxlfoundation/scikit-learn-intelex#2311]
The sigmoid kernel converts data via tanh into a new space. This is easy difficulty, but requires significant benchmarking to find when
the scikit-learn-intelex implementation provides better performance. This project will focus on the public API and including the benchmarking 
results for a seamless, high-performance user experience. Combines with the other kernel projects to a medium time commitment.

https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.sigmoid_kernel.html
